File: Python/1/IsPerm.py
# CareerCup 1.2
# check if str1 is a permutation of str2
#
# @author Chang Liu

def isPerm(str1, str2):
  return sorted(list(str1)) == sorted(list(str2));
  
if __name__ == '__main__':
  print isPerm('abc', 'abc');
  print isPerm('abc', 'abcd');
  print isPerm('abac', 'absc');
  print isPerm(' ', '');
  print isPerm(' ', ' ');

File: Java/1/IsPerm.java
public class IsPerm {
  private char[] str1, str2;
  
  public IsPerm (String str1, String str2) {
    this.str1 = str1.toCharArray();
    this.str2 = str2.toCharArray();
  }
  
  public boolean check() {
    java.util.Arrays.sort(this.str1);
    java.util.Arrays.sort(this.str2);
    return new String(this.str1).equals(new String(this.str2));
  }
  
  public static void main(String args[]) {
    IsPerm is = new IsPerm("abc", "cbaa");
    System.out.println(is.check());
  }
}